src/hotspot/os/aix/porting_aix.cpp

```diff
@@ -244,21 +235,16 @@ bool AixSymbols::get_function_name (
       }
       p_name[i] = '\0';
 
-      // If it is a C++ name, try and demangle it using the Demangle interface (see demangle.h).
-#ifndef DISABLE_DEMANGLE
+      // If it is a C++ name, try and demangle it using the __cxa_demangle interface(see demangle.h).
       if (demangle) {
-        char* rest;
-        Name* const name = Demangle(p_name, rest);
-        if (name) {
-          const char* const demangled_name = name->Text();
-          if (demangled_name) {
-            strncpy(p_name, demangled_name, namelen-1);
-            p_name[namelen-1] = '\0';
+        int status;
+        char *demangled_name = abi::__cxa_demangle(p_name, nullptr, nullptr, &status);
+        if ((demangled_name != nullptr) && (status == 0)) {
```

Comment:
Should the `strncpy` better depend on `status == 0` and the `free` operation on `demangled_name != nullptr`? I'm not sure about the relationship between the two checks. Maybe your version is ok, too.

Reply:
https://gcc.gnu.org/onlinedocs/libstdc++/libstdc++-html-USERS-4.3/a01696.html does not explicitly state an explicit connection between status == 0 and ptr != nullptr. Its at least a bit vague. A bad implementation could allocate the memory and then fail to demangle. So, yes, I'd tie the free to ptr!=NULL only.